src/openzaak/components/autorisaties/api/serializers.py

```diff
@@ -2,18 +2,83 @@
+
 logger = logging.getLogger(__name__)
 
 
 class ApplicatieSerializer(_ApplicatieSerializer):
+    def to_representation(self, instance):
+        """
+        Join the regular `Applicatie.autorisaties` with `CatalogusAutorisaties`, by
+        adding a virtual `Autorisatie` for each zaak/besluit/informatieobjecttype in the
+        linked `Catalogus`
+        """
+        from ..forms import COMPONENT_TO_FIELDS_MAP
+
+        data = super().to_representation(instance)
+
+        virtual_autorisaties = []
+        for catalogus_autorisatie in instance.catalogusautorisatie_set.all():
+            if catalogus_autorisatie.component not in COMPONENT_TO_FIELDS_MAP:
+                continue
+
+            # Get the related zaak/informatieobject/besluittypen related to this Catalogus
+            # (dependent on the component of the CatalogusAutorisatie in the current iteration)
+            type_field = COMPONENT_TO_FIELDS_MAP[catalogus_autorisatie.component][
+                "_autorisatie_type_field"
+            ]
+
+            # Instead of accessing the *type_set using `getattr`, we explicitly use the
+            # dot notation here, because the `getattr` approach means we cannot rely on the
+            # optimization `prefetch_related` brings to the queryset defined on the viewset
+            if catalogus_autorisatie.component == ComponentTypes.zrc:
+                types = catalogus_autorisatie.catalogus.zaaktype_set.all()
+            elif catalogus_autorisatie.component == ComponentTypes.drc:
+                types = catalogus_autorisatie.catalogus.informatieobjecttype_set.all()
+            elif catalogus_autorisatie.component == ComponentTypes.brc:
+                types = catalogus_autorisatie.catalogus.besluittype_set.all()
+
+            virtual_autorisaties += [
+                Autorisatie(
+                    **{
+                        "applicatie": instance,
+                        "component": catalogus_autorisatie.component,
+                        "scopes": catalogus_autorisatie.scopes,
+                        "max_vertrouwelijkheidaanduiding": catalogus_autorisatie.max_vertrouwelijkheidaanduiding,
+                        type_field: build_absolute_url(
+                            reverse(
+                                f"{type_field}-detail",
+                                kwargs={
+                                    "uuid": str(type.uuid),
+                                    "version": settings.REST_FRAMEWORK[
+                                        "DEFAULT_VERSION"
+                                    ],
+                                },
+                            ),
+                            request=self.context["request"],
+                        ),
+                    }
+                )
+                for type in types
+            ]
+
+        serializer = AutorisatieBaseSerializer(virtual_autorisaties, many=True)
+        data["autorisaties"] = data["autorisaties"] + serializer.data
```

Comment:
This question is more related to I guess how the Autorisatie API works but how does it handle situations where there is an autorisatie for a certain catalogus and there is another autorisation for the same component with another `max_vertrouwelijkheidaanduiding`?

Reply:
I'm not sure what you mean, do you mean how this situation affects permission handling, how it affects the Applicaties API endpoint itself or what happens when you save these in the admin form?

With regards to the latter, I checked this an there is indeed an issue here that needs to be fixed, because the CatalogusAutorisatie gets deleted (max VA is not checked when doing this)